runtime/commonMain/src/kotlinx/serialization/SerialDescriptorBuilder.kt

```diff
@@ -28,6 +28,7 @@ import kotlinx.serialization.internal.*
  * }
  * ```
  */
+@Suppress("FunctionName")
```

Comment:
I'd rather suggest you to change the regexp of the inspection

Reply:
It's Kotlin's default inspection, not ours. We may create a ticket and discuss it, since constructor-like functions are widely used (by us at least), but it is not clear how to distinguish constructor function from regular one.